runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4855,6 +4865,15 @@ static TR::Register * generateMultianewArrayWithInlineAllocators(TR::Node *node,
    cursor = generateRXInstruction(cg, use64BitClasses ? TR::InstOpCode::STG : TR::InstOpCode::ST, node, componentClassReg, generateS390MemoryReference(temp2Reg, TR::Compiler->om.offsetOfObjectVftField(), cg));
    iComment("Init 2nd dim class field.");
 
+   // If batch clear is disabled, set element's size and mustBeZero ('0') fields to 0
+   if (disableBatchClear)
+      {
+      cursor = generateRXInstruction(cg, TR::InstOpCode::ST, node, secondDimLenReg, generateS390MemoryReference(temp2Reg, fej9->getOffsetOfContiguousArraySizeField(), cg));
+      iComment("Init 2st dim size field.");
+      cursor = generateRXInstruction(cg, TR::InstOpCode::ST, node, secondDimLenReg, generateS390MemoryReference(temp2Reg, fej9->getOffsetOfDiscontiguousArraySizeField(), cg));
```

Comment:
in this case the first dimension is not zero and in line 4844 we set the size field but we never set `DiscontiguousArraySize` to zero in X86. so maybe we can remove this instruction for zero size as well.